src/index.js

```diff
@@ -60,6 +54,14 @@ function makeHardener(initialFringe, freeze) {
     }
 
     function freezeAndTraverse(obj) {
+      // Naively prepare the object for freezing.
```

Comment:
Now that I know the use case you have in mind, let's discuss `immunize` further before revisiting this PR. I remain skeptical that `immunize` is a good idea. (Great name btw ;).) Example:

```js
function foo() {}
const dag = { foo, foo };
// dag.foo === dag.foo

immunize(dag);
// dag.foo !== dag.foo
```

Even if you complexify `immunize` enough to handle such cases, I remain nervous about trying to do surgery to the object graph in place, especially transitively.

Reply:
Let's continue this in [the Jessie issue](https://github.com/Agoric/Jessie/issues/27).